proto-mailbox/src/main/kotlin/actor/proto/mailbox/DefaultDispatcher.kt

```diff
@@ -1,13 +1,22 @@
 package actor.proto.mailbox
 
-import kotlinx.coroutines.GlobalScope
 import kotlinx.coroutines.launch
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.plus
+import kotlin.coroutines.CoroutineContext
 
 
-class DefaultDispatcher( override var throughput: Int = 300) : Dispatcher {
-    override fun schedule(mailbox:Mailbox) {
+class DefaultDispatcher(context: CoroutineContext = Dispatchers.Default, override var throughput: Int = 300) : Dispatcher {
+    // We could create a jobless GlobalScope root scope and replace the default dispatcher with our dispatcher
+    // val scope : CoroutineScope = GlobalScope + context
+    //Or we can create a scope from our dispatcher with a default job, and replace it with a supervisor job.
+    //This way we get a normal scope that could be cancelled if needed
+    private val scope : CoroutineScope = CoroutineScope(context) + SupervisorJob()
```

Comment:
Do we need the supervisor, because we actually not doing anything with it. I've to admit that my coroutines knowledge is still pretty limited. 


Also the example https://kotlinlang.org/docs/reference/coroutines/exception-handling.html#supervision-job shows the supervisor registration like this: `CoroutineScope(context + SupervisorJob())`

Reply:
I think both a jobless scope and a supervisor job scope will behave the same for our purposes. If in future we wanted to cancel the scope and all its children, then we need a supervisor job scope - but we don't currently do that. I opted for the supervisor job scope, but couldn't see a strong argument, so left the option in the comments.

I think a good future step would be to have a scope created per actor. Then when the actor is terminated, the scope can be cancelled, clearing up any coroutines, and avoiding orphans. But that needs more extensive refactoring.

I think these two are identical:
CoroutineScope(context) + SupervisorJob()
CoroutineScope(context+ SupervisorJob())

See here: https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/plus.html

I opted for the former but I don't feel strongly as to which one is better.